sql/catalog/lease: TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces failed

sql/catalog/lease.TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13159526?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13159526?buildTab=artifacts#/) on release-23.2.0-rc @ [e5f281408027c52fa0f63dd48401a25a61eccfd7](https://github.com/cockroachdb/cockroach/commits/e5f281408027c52fa0f63dd48401a25a61eccfd7):


```
=== RUN   TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/ce8d73b4a4081f39d36a7fa2c2508fcc/logTestRangefeedUpdatesHandledProperlyInTheFaceOfRaces963117018
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:159: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    lease_test.go:2340: test took 32.238806118s which is greater than 30s
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/ce8d73b4a4081f39d36a7fa2c2508fcc/logTestRangefeedUpdatesHandledProperlyInTheFaceOfRaces963117018
--- FAIL: TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces (32.27s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34670

--- a/pkg/sql/catalog/lease/lease_test.go
+++ b/pkg/sql/catalog/lease/lease_test.go
@@ -2349,7 +2349,7 @@ func TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 	defer ensureTestTakesLessThan(t, 30*time.Second)()
-	skip.UnderRace(t, "test must take less than 30 seconds, and race builds are slow")
+	skip.UnderDuress(t, "test must take less than 30 seconds, so avoid slow configs")
 
 	ctx := context.Background()
 	var interestingTable atomic.Value

